### DISCUSSION TITLE
No profiles for <one_signal_extension_bundle_id> were found: Xcode couldn't find any iOS App Development provisioning profiles matching <one_signal_extension_bundle_id>

### DISCUSSION THREAD

**Discussion by cheeweichoy88** [codemagic-ci-cd/codemagic-docs#1798] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I have followed the instruction in https://docs.codemagic.io/flutter-publishing/publishing-to-app-store/
 - Step 1. Creating an App Store API key for Codemagic
 - Step 2. Connecting the Apple Developer Portal integration for your team/account 
 - Step 3. Enabling App Store Connect publishing for workflow
 
 after that i encounter an error when building ios app with codemagic
 

> Xcode archive done.                                          2.7s
> Failed to build iOS app
> Error (Xcode): No profiles for <one_signal_extension_bundle_id>  were found: Xcode couldn't find any iOS App Development provisioning profiles matching <one_signal_extension_bundle_id>  Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
> /Users/builder/clone/ios/Runner.xcodeproj

> Error (Xcode): No profiles for <my_bundle_id> were found: Xcode couldn't find any iOS App Development provisioning profiles matching <my_bundle_id>. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
> /Users/builder/clone/ios/Runner.xcodeproj
 
> It appears that there was a problem signing your application prior to installation on the device.

> Verify that the Bundle Identifier in your project is your signing id in Xcode
>  open ios/Runner.xcworkspace

> Also try selecting 'Product > Build' to fix the problem.
> Encountered error while archiving for device.

my workflow settings

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/137255808/96ea3b2b-b2b8-41f3-a986-f80b333114ff)

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/137255808/8ea25cbb-08fd-4cf9-b0d2-d8f6a3ed5ea2)

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/137255808/032a0239-5c23-46a9-b8f7-3fb937b4ba1b)

**ps: I'm able to build and release my ios build to test flight, but encounter this error while building ios with codemagic**

### Expected results

able to build and release my ios app to test flight

### Actual results

unable to build and release my ios app to test flight

**Comment by himesh-cm** [comment]
Hello @cheeweichoy88, kindly take note that if your app has app extensions, it is necessary to upload an extra provisioning profile for every extension. May I request a build id so I can investigate it more thoroughly?

**Reply by cheeweichoy88** [reply]
build id  : 64916c6605a53d9963ce962b

yes, i did add an extra provisioning profile for my extension, you can refer to my second image above

**Reply by himesh-cm** [reply]
Hi @cheeweichoy88, I see that the latest build was successful. Can you confirm is this is still an issue for you?

**Reply by himesh-cm** [reply]
Please not that your team is on a paid plan which means that you should have access to the private chat widget in the UI. Please use that next time for quicker responses.

**Reply by himesh-cm** [reply]
The conversation will continue on intercom. I will update the resolution here once it's fixed.

**Reply by himesh-cm** [reply]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
